### PR TITLE
Add two newlines between versions to make them stand out more.

### PR DESCRIPTION
**Pull request by skorokithakis**

### CODE REVIEW

**Comment by vaab** [comment]
Hi, thanks for you interest in gitchangelog.

Format discussions are always a little bit subjective. Although, I guess I could do with 1 space more (to mimick pep8 on class compared to methods). There are tests to modify so that they pass, and the others implementations to update, also.

Thanks for your contribution.

**Comment by skorokithakis** [comment]
Sure. How do I run the tests? There's nothing in the documentation, apart from a `.travis.yml` file.

**Comment by vaab** [comment]
Indeed, as found in ``.travis.yml`` file, it's:

    nosetests .

**Comment by skorokithakis** [comment]
Alright, this took me an ungodly amount of time (due to tests being weird and failing), but it's finally passing all the tests. Can it be merged?

**Comment by vaab** [comment]
Thanks for your interest and this PR. It seems I'll have to reformat the commit history so it follows gitchangelog's conventions.

I see also that there was a misunderstanding (my bad) about the number of spaces before sections. I thought you wanted in total 2 new lines to mimick pep8 with classes opposed to method. Why not before versions section then ?

Your implementation on mako and mustache version *outside* templates forces unacceptable changes outside the templates that might break existing other templates. You should be able to achieve what you want by only tampering with the templates.

I have put all commit in one, and proposed this based on your commits:
https://github.com/vaab/gitchangelog/tree/skorokithakis-patch-1
(check df9eac354051e58815eb25bdf13271c3dfbc0bad ) 

Would this be okay for you ?

**Comment by skorokithakis** [comment]
That's ideal, thank you. I spent *way* too much time getting tests to run that I got frustrated at such a simple change taking such a long time, and hacked it in the code. Your branch looks ideal, thanks.

**Comment by skorokithakis** [comment]
Also, about the spaces: In general, when rendered to HTML from markdown/rst, everything is fine because of the h1/h2, but in plaintext (which is also often read) the changelog is currently very flat.

Two newlines between sections would also be great (since that would separate them more), but then it seems to me that an extra newline between version sections (i.e. three newlines) would make the changelog even easier to scan.

Basically, my rationale is that there should be better visual separation between all the sections (version- and non-) in the changelog, to make it much easier to visually scan by version or section.

**Comment by vaab** [comment]
ok, I'll have a look at 3-2-1 new lines before Version, Section, Commit. I'm concerned that it might be too much spaces.

**Comment by skorokithakis** [comment]
Yes, you're right. Maybe it can be 2-1-0? The list format is already very readable. Something like this?:

```
v0.0.11 (2016-11-16)
--------------------

Features
~~~~~~~~
- Add music mode. [Stavros Korokithakis]
- Ignore errors on init files. [Stavros Korokithakis]
- Make `ensure_on` more accurate by always fetching the properties
  before a method call. [Stavros Korokithakis]

Fixes
~~~~~
- Add set_adjust. [Stavros Korokithakis]
- Add value parameter to set_hsv. [Stavros Korokithakis]

v0.0.10 (2016-11-15)
--------------------

Features
~~~~~~~~
- Add set_adjust. [Stavros Korokithakis]
- Add value parameter to set_hsv. [Stavros Korokithakis]

Fixes
~~~~~
- Don't require flake8 for tests any more. [Stavros Korokithakis]
- Fix tests on Python 3. [Stavros Korokithakis]
- Fix per-call effects. [Stavros Korokithakis]
```

This is the most readable of all the formats I've tried, while still not having too many spaces. The indentation in lists makes them very readable as well.

**Comment by vaab** [comment]
Thanks for your example, that makes things much clearer. You've forgotten the 2 lines before versions if I understood clearly.
I thought I didn't have the choice for the current one blank line between list elements (I thought it was required by the RST syntax), but I did a quick rendering test and this seems okay.
I'll do some more testing tomorrow (it's late here), and if 2-1-0 doesn't break anything, I'll go for it.

**Comment by skorokithakis** [comment]
Thank you! Hmm, where have I forgotten the lines? There are two newlines before `v0.0.10 (2016-11-15)`. Also lists [in the Sphinx reference](http://www.sphinx-doc.org/en/stable/rest.html) seem perfectly happy with no newlines between them (I think they rely on indentation).

**Comment by vaab** [comment]
Ok, there's the 2-1-0 (see last commit), please note that commit body must have a single new line to separate it from the commit summary, so this is an extensive example taken from gitchangelog's own history.

```
2.2.0 (2015-01-27)
------------------

New
~~~
- Provide support for older config file format. [Valentin Lab]
- Added 'octobercms-plugin' mako template. (fixes #16) [Valentin Lab]
- Added ``body_process`` and ``subject_process`` options. (fixes #22)
  [Valentin Lab]

  These options superseeds ``replace_regexps`` and ``body_split_regexp``
  as they provide a full control over text transformation of the subject
  or the body of the commit before they get included in the changelog.
- Added ``include_merge`` option to filter out merge commit. [Casey
  Duquette]

Changes
~~~~~~~
- Produce a more linear commit history (fixes #14) [Casey Duquette]

  Instead of retrieving the git log ordered by date, retrieve the log as
  a difference between tags to produce a more accurate view of changes
  between releases.

  For instance, imagine this git graph::

    * 6c0fd62 (HEAD, tag: sprint-6, origin/smoke, smoke, develop)
    *   5292a28 Merge back to develop
    |\
    | * 6612fce (tag: sprint-5.1, origin/master, origin/HEAD, master) super important hotfix
    * | 7d6286f more development work
    * | 8c1e3d6 continued development work
    * | fa3d4bd development work
    |/
    * ec1a19c (tag: sprint-5)

  Previously, commits ``fa3d4bd``, ``8c1e3d6``, ``7d6286f`` that
  occurred on the develop branch before the hotfix that led to tagging
  ``sprint-5.1``, were captured in the changelog under release
  ``sprint-5.1`` because of the order of the commits. But it is obvious
  that these commits were not included in a release until
  ``sprint-6``. The new method of calculating the changelog will capture
  this and reflect it properly, assigning those changes to ``sprint-6``.

Fix
~~~
- Last commit was omitted (fixes #23). [Valentin Lab]
- Bogus messages when template didn't exist. [Valentin Lab]

  Refactored out the common code and corrected the bad error message.
- Removed hypothetical memory exhaust while parsing ``git log``.
  [Valentin Lab]

  Parse stdout as it's produced by git log by chunks.


2.1.2 (2014-04-25)
------------------

Fix
~~~
- Fail with error message when config path exists but is not a file.
  (fixes #11) [Casey Duquette]

  For example, the config file could be a directory.


2.1.1 (2014-04-15)
------------------

Fix
~~~
- Removed exception if you had file which name that matched a tag's
  name. (fixes #9) [Valentin Lab]

**Comment by skorokithakis** [comment]
Looks fantastic, thank you! I see you've also released it already? Great job!
